### ISSUE TITLE
Xaero's minimap 24.x.x compatibility issue

### ISSUE THREAD

**Issue by hatkidchan** [lanAnderson/SeasonInfo#73]
**Describe the issue**
- Crash when joining the world with Xaero's minimap 24.0.2
- On seasonhud 1.7.13 w/ Xaero's minimap 23.9.7 it works fine. I guess last major release of Xaero's minimap broke it.

**Minecraft Version**
- 1.20.1

**Mod Version**
- 1.7.14

**Modloader(Fabric/Forge/NeoForge)**
- Fabric

**Modloader Version**
- 0.15.7

**Additional Info (Crashlog/Modlist)**
<details>
<summary>Relevant part of the log</summary>

```
[14:10:07] [Render thread/ERROR]: Unreported exception thrown!
java.lang.NoSuchMethodError: 'xaero.common.AXaeroMinimap xaero.common.XaeroMinimapSession.getModMain()'
	at club.iananderson.seasonhud.impl.minimaps.HiddenMinimap.minimapHidden(HiddenMinimap.java:23) ~[seasonhud-fabric-1.20.1-1.7.14.jar:?]
	at club.iananderson.seasonhud.client.SeasonHUDOverlay.onHudRender(SeasonHUDOverlay.java:58) ~[seasonhud-fabric-1.20.1-1.7.14.jar:?]
	at net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback.lambda$static$0(HudRenderCallback.java:27) ~[fabric-rendering-v1-3.0.8+1802ada577-dd09b4eef5150511.jar:?]
	at net.minecraft.class_329.handler$dbb002$fabric-rendering-v1$render(class_329.java:10386) ~[client-intermediary.jar:?]
	at net.minecraft.class_329.method_1753(class_329.java:370) ~[client-intermediary.jar:?]
	at net.minecraft.class_757.method_3192(class_757.java:926) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1219) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:802) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[minecraft-1.20.1-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.7.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.7.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.7.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) ~[NewLaunch.jar:?]
[14:10:07] [Render thread/INFO]: Minimap session finalized.
[14:10:07] [Render thread/INFO]: Finalizing world map session...
[14:10:07] [Thread-10/INFO]: World map force-cleaned!
[14:10:07] [Render thread/INFO]: World map session finalized.
[14:10:08] [Render thread/INFO]: Cleared all avatars
[14:10:08] [Render thread/INFO]: Clearing 'BetterStatsClientNetworkHandler' flags.
[14:10:08] [Render thread/INFO]: Cleared all avatars
[14:10:08] [Render thread/INFO]: Clearing 'BetterStatsClientNetworkHandler' flags.
[14:10:08] [Render thread/INFO]: OpenGL debug message: id=1, source=API, type=ERROR, severity=HIGH, message='GL_INVALID_ENUM in glPolygonMode(face)'
[14:10:08] [Render thread/INFO]: OpenGL debug message: id=1, source=API, type=ERROR, severity=HIGH, message='GL_INVALID_ENUM in glPolygonMode(face)'
[14:10:08] [Render thread/FATAL]: Unreported exception thrown!
java.lang.NoSuchMethodError: 'xaero.common.AXaeroMinimap xaero.common.XaeroMinimapSession.getModMain()'
	at club.iananderson.seasonhud.impl.minimaps.HiddenMinimap.minimapHidden(HiddenMinimap.java:23) ~[seasonhud-fabric-1.20.1-1.7.14.jar:?]
	at club.iananderson.seasonhud.client.SeasonHUDOverlay.onHudRender(SeasonHUDOverlay.java:58) ~[seasonhud-fabric-1.20.1-1.7.14.jar:?]
	at net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback.lambda$static$0(HudRenderCallback.java:27) ~[fabric-rendering-v1-3.0.8+1802ada577-dd09b4eef5150511.jar:?]
	at net.minecraft.class_329.handler$dbb002$fabric-rendering-v1$render(class_329.java:10386) ~[client-intermediary.jar:?]
	at net.minecraft.class_329.method_1753(class_329.java:370) ~[client-intermediary.jar:?]
	at net.minecraft.class_757.method_3192(class_757.java:926) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1219) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:802) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[minecraft-1.20.1-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.7.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.7.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.7.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) ~[NewLaunch.jar:?]
[14:10:08] [Render thread/FATAL]: Minecraft ran into a problem! Report saved to: /home/hkc/.local/share/PrismLauncher/instances/Diacraft-staging/.minecraft/crash-reports/crash-2024-03-19_14.10.08-client.txt
---- Minecraft Crash Report ----
// Why is it breaking :(

Time: 2024-03-19 14:10:08
Description: Unexpected error

java.lang.NoSuchMethodError: 'xaero.common.AXaeroMinimap xaero.common.XaeroMinimapSession.getModMain()'
	at club.iananderson.seasonhud.impl.minimaps.HiddenMinimap.minimapHidden(HiddenMinimap.java:23)
	at club.iananderson.seasonhud.client.SeasonHUDOverlay.onHudRender(SeasonHUDOverlay.java:58)
	at net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback.lambda$static$0(HudRenderCallback.java:27)
	at net.minecraft.class_329.handler$dbb002$fabric-rendering-v1$render(class_329.java:10386)
	at net.minecraft.class_329.method_1753(class_329.java:370)
	at net.minecraft.class_757.method_3192(class_757.java:926)
	at net.minecraft.class_310.method_1523(class_310.java:1219)
	at net.minecraft.class_310.method_1514(class_310.java:802)
	at net.minecraft.client.main.Main.main(Main.java:250)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at club.iananderson.seasonhud.impl.minimaps.HiddenMinimap.minimapHidden(HiddenMinimap.java:23)
	at club.iananderson.seasonhud.client.SeasonHUDOverlay.onHudRender(SeasonHUDOverlay.java:58)
	at net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback.lambda$static$0(HudRenderCallback.java:27)
	at net.minecraft.class_329.handler$dbb002$fabric-rendering-v1$render(class_329.java:10386)
	at net.minecraft.class_329.method_1753(class_329.java:370)

-- Affected level --
Details:
	All players: 1 total; [class_746['hatkidchan'/172, l='ClientLevel', x=-19.12, y=112.84, z=-249.53]]
	Chunk stats: 729, 165
	Level dimension: minecraft:overworld
	Level spawn location: World: (0,95,0), Section: (at 0,15,0 in 0,5,0; chunk contains blocks 0,-64,0 to 15,319,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 113332731 game time, 1779485 day time
	Server brand: fabric
	Server type: Non-integrated multiplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:458)
	at net.minecraft.class_310.method_1587(class_310.java:2406)
	at net.minecraft.class_310.method_1514(class_310.java:826)
	at net.minecraft.client.main.Main.main(Main.java:250)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
```
</details>

**Comment by hatkidchan** [comment]
My suspicion is confirmed by looking at the [changelog for 24.0.1](https://chocolateminecraft.com/update.php?mod_id=0)

>  (1.20.4 - 1.18.2) Restructured my development setup to make future updates faster to develop for multiple mod loaders. Mentioning this because it might have introduced new bugs. Unfortunately this also means that the new minimap/world map releases for Fabric no longer support older versions of the other mod (e.g. latest world map with older minimap). The same changes on Fabric should also break compatibility with most third party mods that reference this mod's code. (Neo)Forge should be fine but I'm planning to make much bigger structural changes in the future now that I'm finally able to. If you're developing a third party mod that fetches the minimap instance from the AXaeroMinimap class, you should now use HudMod.INSTANCE instead.

**Comment by lanAnderson** [comment]
Thanks, I'll take a look today and build against that new version

**Comment by lanAnderson** [comment]
Should be fixed in 1.7.15

**Comment by hatkidchan** [comment]
wow that was quick, thank you!
